### PR TITLE
Update deployment name and labels

### PR DESCRIPTION
**Pull request by ajaypratap003**
## Description

- Update deployment name `rhoai-operator-controller-manager`
- Add new label name: rhoai-operator

- Fix e2e-tests

- https://issues.redhat.com/browse/RHOAIENG-3901
- https://issues.redhat.com/browse/RHOAIENG-3900

## Merge criteria

- [x] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [x] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [x] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by zdtsw** [comment]
i do not see any requirement saying `redhat-ods `need to be updated   => any reason to change this 
the only thing i know is `rhods` should be replaced by `rhoai`

**Comment by ajaypratap003** [comment]
> i do not see any requirement saying `redhat-ods `need to be updated => any reason to change this the only thing i know is `rhods` should be replaced by `rhoai`

I don't have strong opinion. @VaishnaviHire suggested that we should also change deployment name in `rhods-operator` repo as well. 
@VaishnaviHire wdyt ?

**Comment by zdtsw** [comment]
> > i do not see any requirement saying `redhat-ods `need to be updated => any reason to change this the only thing i know is `rhods` should be replaced by `rhoai`
> 
> I don't have strong opinion. @VaishnaviHire suggested that we should also change deployment name in `rhods-operator` repo as well. @VaishnaviHire wdyt ?

ok, if this is to match nightly i am fine

**Review by asanzgom — APPROVED** [review]

File: bundle/manifests/rhoai-operator-webhook-service_v1_service.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v1
```
Comment (ykaliuta) [inline bundle/manifests/rhoai-operator-webhook-service_v1_service.yaml:1]: There is a question about this file https://github.com/red-hat-data-services/rhods-operator/pull/361#issuecomment-2365226293

File: bundle/manifests/rhoai-operator-metrics-reader_rbac.authorization.k8s.io_v1_clusterrole.yaml [hunk 2]
```diff
@@ -2,15 +2,9 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
   creationTimestamp: null
-  name: prometheus-k8s-viewer
+  name: rhoai-operator-metrics-reader
```
Comment (VaishnaviHire) [inline bundle/manifests/rhoai-operator-metrics-reader_rbac.authorization.k8s.io_v1_clusterrole.yaml:5]: Any reason this clusterrole is updated?

Comment (ajaypratap003) [reply bundle/manifests/rhoai-operator-metrics-reader_rbac.authorization.k8s.io_v1_clusterrole.yaml:5]: I didn't update this manually. It updated by deployment name change after running `make bundle` cmd.

**Comment by VaishnaviHire** [comment]
This PR should fix unnecessary bundle updates https://github.com/red-hat-data-services/rhods-operator/pull/370

**Comment by VaishnaviHire** [comment]
@ajaypratap003 Just an fyi we need to update deployment here https://gitlab.cee.redhat.com/data-hub/rhods-cpaas-midstream/-/merge_requests/2795 to match the one in this repo `rhoai-operator-controller-manager`

File: bundle/manifests/rhoai-operator-webhook-service_v1_service.yaml (hunk 1 above)
Comment (ajaypratap003) [reply bundle/manifests/rhoai-operator-webhook-service_v1_service.yaml:1]: @VaishnaviHire Can you please confirm that we should keep this file or not in PR ?

**Comment by zdtsw** [comment]
close PR as we will not need to change in this repo

**Pull request closed without merging**